posthog/src/test/java/com/posthog/java/PostHogTest.java

```diff
@@ -0,0 +1,133 @@
+package com.posthog.java;
+
+import static org.junit.Assert.assertEquals;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneId;
+
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+
+import mockit.Mock;
+import mockit.MockUp;
+
+public class PostHogTest {
+    private PostHog ph;
+    private TestSender sender;
+    private QueueManager queueManager;
+    private String instantExpected = "2020-02-02T02:02:02Z";
+    private Clock clock;
+
+    private void updateInstantNow(String instant) {
+        instantExpected = instant;
+        clock = Clock.fixed(Instant.parse(instantExpected), ZoneId.of("UTC"));
+    }
+
+    @Before
+    public void setUp() {
+        updateInstantNow(instantExpected);
+        new MockUp<Instant>() {
+            @Mock
+            public Instant now() {
+                return Instant.now(clock);
+            }
+        };
+        sender = new TestSender();
+        // by default not sleeping just dependent on queue size of 1, i.e. each call
+        // separately
+        queueManager = new QueueManager.Builder(sender).sleepMs(0).maxTimeInQueue(Duration.ofDays(5)).maxQueueSize(1)
```

Comment:
Is this the best way to structure things in the Java world? I'm just so used to config objects that it looks a bit messy to have to call separate methods for each option I want to set.

Also what happens if we start adding more and more options?

Do feel free to ignore though if this is standard.

Reply:
Can confirm. This is a pattern in Java as ugly as it looks...